src/test/java/io/appium/java_client/service/local/ServerBuilderTest.java

```diff
@@ -1,311 +1,301 @@
-        file = new File("Log3.txt");
-        
-        service = new AppiumServiceBuilder()
-                .withArgument(() -> "-g", file.getAbsolutePath())
-                .build();
-        service.start();
-        assertTrue(file.exists());
-    }
-}
+package io.appium.java_client.service.local;
+
+import static io.appium.java_client.remote.AndroidMobileCapabilityType.APP_ACTIVITY;
+import static io.appium.java_client.remote.AndroidMobileCapabilityType.APP_PACKAGE;
+import static io.appium.java_client.remote.AndroidMobileCapabilityType.CHROMEDRIVER_EXECUTABLE;
+import static io.appium.java_client.remote.MobileCapabilityType.APP;
+import static io.appium.java_client.remote.MobileCapabilityType.FULL_RESET;
+import static io.appium.java_client.remote.MobileCapabilityType.NEW_COMMAND_TIMEOUT;
+import static io.appium.java_client.remote.MobileCapabilityType.PLATFORM_NAME;
+import static io.appium.java_client.service.local.AppiumDriverLocalService.buildDefaultService;
+import static io.appium.java_client.service.local.AppiumServiceBuilder.APPIUM_PATH;
+import static io.appium.java_client.service.local.flags.GeneralServerFlag.CALLBACK_ADDRESS;
+import static io.appium.java_client.service.local.flags.GeneralServerFlag.PRE_LAUNCH;
+import static io.appium.java_client.service.local.flags.GeneralServerFlag.SESSION_OVERRIDE;
+import static java.lang.System.getProperty;
+import static java.lang.System.setProperty;
+import static java.util.Arrays.asList;
+import static java.util.Optional.ofNullable;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import org.apache.commons.validator.routines.InetAddressValidator;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.util.Enumeration;
+import java.util.List;
+
+public class ServerBuilderTest {
+
+    /**
+     * It may be impossible to find the path to the instance of appium server due to different circumstance.
+     * So user may use environment variables/system properties to define the full path to the server
+     * appium.js that is supposed to be default.
+     */
+    private static final String PATH_TO_APPIUM_NODE_IN_PROPERTIES = getProperty(APPIUM_PATH);
+
+    private static final String ROOT_TEST_PATH = "src/test/java/io/appium/java_client/";
+
+    /**
+     * This is the path to the stub appium.js file
+     */
+    private static final String PATH_T0_TEST_APPIUM_JS = ROOT_TEST_PATH + "service/local/appium.js";
```

Comment:
It will be more handy to keep it as File object:
`TEST_APPIUM_JS = Path.get(ROOT_TEST_PATH).resolve("service").resolve("local").resolve("appium.js").toFile()`

Reply:
@TikhomirovSergey Why don't we name stub as `main.js` in sync with appium server's `main.js`?

Reply:
@SrinivasanTarget 
Because it searches for appium.js when it tries to launch the node locally

Reply:
@TikhomirovSergey with the latest appium its main.js

Reply:
https://github.com/appium/java-client/blob/master/src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java#L70

@SrinivasanTarget @saikrishna321 Oops! My fault. yes. Guys you are right :)